content/en/blog/_posts/2022-04-12-volume-populators-beta.md

```diff
@@ -0,0 +1,159 @@
+
+## New Features
+
+The [lib-volume-populator](https://github.com/kubernetes-csi/lib-volume-populator) library
+on which populators are built now includes metrics to help operators monitor and detect
+problems. This library is now beta and latest release is v1.0.1.
+
+The [volume data source validator](https://github.com/kubernetes-csi/volume-data-source-validator)
+controller also has metrics support added, and is in beta. The `VolumePopulator` CRD is
+beta and the latest release is v1.0.1.
+
+## Trying it out
+
+To see how this works, you can install the sample "hello" populator and try it
+out.
+
+First install the volume-data-source-validator controller.
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/volume-data-source-validator/v1.0.1/client/config/crd/populator.storage.k8s.io_volumepopulators.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/volume-data-source-validator/v1.0.1/deploy/kubernetes/rbac-data-source-validator.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/volume-data-source-validator/v1.0.1/deploy/kubernetes/setup-data-source-validator.yaml
+```
+
+Next install the example populator.
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/lib-volume-populator/v1.0.1/example/hello-populator/crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/lib-volume-populator/master/example/hello-populator/deploy.yaml
+```
+
+Your cluster now has a new CustomResourceDefinition that provides a test API named Hello.
+Create an instance of the `Hello` custom resource, with some text:
+
+```yaml
+apiVersion: hello.example.com/v1alpha1
+kind: Hello
+metadata:
+  name: example-hello
+spec:
+  fileName: example.txt
+  fileContents: Hello, world!
+```
+
+Create a PVC that refers to that CR as its data source.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: example-pvc
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Mi
+  dataSourceRef:
+    apiGroup: hello.example.com
+    kind: Hello
+    name: example-hello
+  volumeMode: Filesystem
+```
+
+Next, run a Job that reads the file in the PVC.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: example-job
+spec:
+  template:
+    spec:
+      containers:
+        - name: example-container
+          image: busybox:latest
+          command:
+            - cat
+            - /mnt/example.txt
+          volumeMounts:
+            - name: vol
+              mountPath: /mnt
+      restartPolicy: Never
+      volumes:
+        - name: vol
+          persistentVolumeClaim:
+            claimName: example-pvc
+```
+
+Wait for the job to complete (including all of its dependencies).
+
+```shell
+kubectl wait --for=condition=Complete job/example-job
+```
+
+And last examine the log from the job.
+
+```shell
+kubectl logs job/example-job
+Hello, world!
```

Comment:
It's best not to include the output in the box: `Hello, world!` isn't shell that people should run. Could you start a new block (see [Separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output) for an example)